woocommerce/woocommerce-android
# All Product Reviews: tracks events

Fixes #1421 by adding the following tracks events:

- `REVIEWS_LOADED`
- `REVIEWS_LOAD_FAILED`
- `REVIEWS_PRODUCTS_LOADED`
- `REVIEWS_PRODUCTS_LOAD_FAILED`
- `REVIEW_LOADED`
- `REVIEW_LOAD_FAILED`
- `REVIEW_PRODUCT_LOADED`
- `REVIEW_PRODUCT_LOAD_FAILED`
- `REVIEW_MARK_READ`
- `REVIEW_MARK_READ_SUCCESS`
- `REVIEW_MARK_READ_FAILED`
- `REVIEWS_MARK_ALL_READ`
- `REVIEWS_MARK_ALL_READ_SUCCESS`
- `REVIEWS_MARK_ALL_READ_FAILED`
- `NOTIFICATIONS_LOAD_FAILED`
- `REVIEW_LIST_PULLED_TO_REFRESH`
- `REVIEWS_LIST_MENU_MARK_READ_BUTTON_TAPPED`

As well as fixing how the `NOTIFICATION_OPEN` is fired. Now it will only be fired if the Product Review has a matching notification. It was also added to be fired when an Order is opened that has a matching notification.

## Testing

An easy way to test without having to make a bunch of new product reviews is to replace [this line](https://github.com/woocommerce/woocommerce-android/blob/46b195f4a5bbf1c89bfbc470efcd789e2d0bb9db/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewListFragment.kt#L138) with this:
```
viewModel.reloadReviewsFromCache()
viewModel.checkForUnreadReviews()
isRefreshing = false
```
Then:
1. Build and run the app
2. Select the reviews tab and wait for it to finish refreshing.
3. Execute this SQL statement against the device's local database:
```
UPDATE NotificationModel SET read = 0
```
_(Can be done either using terminal to manually run the statement directly on the device database, or, if you have SQLiteExplorer installed in Android Studio run it inside the Console using the Live Database Connection)_

4. All the reviews should now show up as unread. 

## TODO 

- [x] Add tracks events to the spreadsheet
- [x] Register and validate new tracks events
